lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
I can move the assertion to line 51. so that it is after 'find_service_plist' but before 'find_service_label'